### PR TITLE
DEV-2362: Fix donate button label, prevent submission when no amount

### PR DESCRIPTION
**Pull request by nrh-cklimas**
Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)

n/a

#### What's this PR do?

Fixes a bug where NaN shows up in the donation form button label, and the form can mistakenly submitted.

#### Why are we doing this? How does it help us?

Improves donation form validation.

#### How should this be manually tested? Please include detailed step-by-step instructions.

1. Go to a donation page.
2. Click on the “free form” amount input, but don’t enter any text.
3. Now click on the “Agree to pay fees” button and toggle it to off position.
4. Scroll down to the submit button. You should see that the text “Enter a valid amount" is in the label and it looks disabled.
5. Click the button and confirm the form is not submitted.

#### Did this PR make changes to the user interface that may require changes to the user-facing docs?

No.

#### Have automated unit tests been added? If not, why?

Added a Cypress test to cover this bug and a couple other user interactions that might trigger similar problems.

#### How should this change be communicated to end users?

n/a

#### Are there any smells or added technical debt to note?

n/a

#### Has this been documented? If so, where?

No.

#### What are the relevant tickets? Add a link to any relevant ones.

https://news-revenue-hub.atlassian.net/browse/DEV-2362

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?

No.

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:

No.

### CODE REVIEW

**Review by GuiMend — APPROVED** [review]
Have one comment, besides that it's good

File: spa/cypress/integration/01 - donation-page.spec.js [hunk 1]
```diff
@@ -602,6 +602,26 @@ describe('User flow: unhappy paths', () => {
...
+
+    cy.getByTestId('amount-other').click();
+    cy.getByTestId('donation-page-submit').should('have.attr', 'disabled');
+    cy.getByTestId('donation-page-submit').should('have.text', 'Enter a valid amount');
+    cy.getByTestId('pay-fees').click();
+    cy.getByTestId('donation-page-submit').should('have.attr', 'disabled');
+    cy.getByTestId('donation-page-submit').should('have.text', 'Enter a valid amount');
+    cy.getByTestId('pay-fees').click();
+    cy.getByTestId('donation-page-submit').should('have.attr', 'disabled');
+    cy.getByTestId('donation-page-submit').should('have.text', 'Enter a valid amount');
```
[the remark this answers is not in the document]
Comment (nrh-cklimas) [reply spa/cypress/integration/01 - donation-page.spec.js:619]: The idea here was to toggle the pay fees checkbox through both states and verify that it doesn't trigger the NaN bug in either case, but it felt kind of ugly to write. Any suggestions on making this better? I couldn't figure out why, but I was unable to chain the Cypress assertions, e.g. `cy.getByTestId('donation-page-submit').should('have.attr', 'disabled').should('have.text', 'Enter a valid amount');`

[the remark this answers is not in the document]
Comment (GuiMend) [reply spa/cypress/integration/01 - donation-page.spec.js:619]: Got it, no problem. Maybe create a looping function to run twice? 🤔 
Would clean up some code and make it more direct that you want to run it twice, but don't know if it's best practice or not

**Review by GuiMend — APPROVED** [review]

File: spa/cypress/integration/01 - donation-page.spec.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (nrh-cklimas) [reply spa/cypress/integration/01 - donation-page.spec.js]: I'm not sure either but a loop makes sense. Updated in f679f6c1bae0801c3915dcb3e249d66b4eca66e4

**Review by x110dc — APPROVED** [review]

**Pull request merged**